test: functional test failures under `--usecli`

Ubuntu 24.04.2 LTS
Python 3.12.3
master @ 817edfb21e4fa96289d69d4e1dbabb6b9ef9d8f5
```bash
cmake -B build
cmake --build build
./build/test/functional/test_runner.py --usecli
```

```bash
feature_assumeutxo.py                                                | ✖ Failed  | 11 s
feature_bip68_sequence.py                                            | ✖ Failed  | 58 s
feature_fastprune.py                                                 | ✖ Failed  | 1 s
feature_fee_estimation.py                                            | ✖ Failed  | 1 s
feature_nulldummy.py                                                 | ✖ Failed  | 1 s
feature_reindex_readonly.py                                          | ✖ Failed  | 0 s
feature_signet.py                                                    | ✖ Failed  | 1 s
feature_taproot.py                                                   | ✖ Failed  | 6 s
mempool_package_rbf.py                                               | ✖ Failed  | 22 s
mining_mainnet.py                                                    | ✖ Failed  | 1 s
p2p_net_deadlock.py --v1transport                                    | ✖ Failed  | 1 s
p2p_net_deadlock.py --v2transport                                    | ✖ Failed  | 1 s
p2p_tx_download.py                                                   | ✖ Failed  | 34 s
rpc_deprecated.py                                                    | ✖ Failed  | 1 s
rpc_estimatefee.py                                                   | ✖ Failed  | 1 s
rpc_generate.py                                                      | ✖ Failed  | 1 s
rpc_getblockfrompeer.py                                              | ✖ Failed  | 1 s
rpc_getdescriptorinfo.py                                             | ✖ Failed  | 1 s
rpc_invalid_address_message.py                                       | ✖ Failed  | 2 s
rpc_packages.py                                                      | ✖ Failed  | 5 s
rpc_scanblocks.py                                                    | ✖ Failed  | 1 s
rpc_scantxoutset.py                                                  | ✖ Failed  | 5 s
rpc_whitelist.py                                                     | ✖ Failed  | 1 s
wallet_bumpfee.py --descriptors                                      | ✖ Failed  | 10 s
wallet_descriptor.py --descriptors                                   | ✖ Failed  | 7 s
wallet_encryption.py --descriptors                                   | ✖ Failed  | 6 s
wallet_fundrawtransaction.py --descriptors                           | ✖ Failed  | 12 s
wallet_labels.py --descriptors                                       | ✖ Failed  | 2 s
wallet_miniscript.py --descriptors                                   | ✖ Failed  | 2 s
wallet_send.py --descriptors                                         | ✖ Failed  | 1 s
```

```bash
147/318 - rpc_deprecated.py failed, Duration: 1 s

stdout:
2025-04-14T09:29:49.903000Z TestFramework (INFO): PRNG seed is: 5746030027211002162
2025-04-14T09:29:49.903000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20250414_092629/rpc_deprecated_169
2025-04-14T09:29:50.177000Z TestFramework (INFO): Tests for deprecated RPC methods (if any)
2025-04-14T09:29:50.178000Z TestFramework (INFO): Tests for deprecated wallet-related RPC methods (if any)
2025-04-14T09:29:50.178000Z TestFramework (INFO): Test settxfee RPC deprecation
2025-04-14T09:29:50.203000Z TestFramework (ERROR): Unexpected exception caught during testing
Traceback (most recent call last):
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 182, in main
    self.run_test()
  File "/root/ci_scratch/build/test/functional/rpc_deprecated.py", line 37, in run_test
    assert_raises_rpc_error(-32, 'settxfee is deprecated and will be fully removed in v31.0.', self.nodes[0].rpc.settxfee, 0.01)
                                                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'settxfee'
```

```bash
131/318 - rpc_whitelist.py failed, Duration: 1 s

stdout:
2025-04-14T09:29:42.390000Z TestFramework (INFO): PRNG seed is: 3309837604605811676
2025-04-14T09:29:42.390000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20250414_092629/rpc_whitelist_186
2025-04-14T09:29:43.102000Z TestFramework (INFO): [user1]: Testing a non permitted permission (getnetworkinfo)
2025-04-14T09:29:43.102000Z TestFramework (ERROR): Unexpected exception caught during testing
Traceback (most recent call last):
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 182, in main
    self.run_test()
  File "/root/ci_scratch/build/test/functional/rpc_whitelist.py", line 78, in run_test
    assert_equal(403, rpccall(self.nodes[0], user, permission).status)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/ci_scratch/build/test/functional/rpc_whitelist.py", line 20, in rpccall
    conn = http.client.HTTPConnection(url.hostname, url.port)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/http/client.py", line 880, in __init__
    (self.host, self.port) = self._get_hostport(host, port)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/http/client.py", line 926, in _get_hostport
    i = host.rfind(':')
        ^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'rfind'
```

```bash
90/318 - feature_bip68_sequence.py failed, Duration: 58 s

stdout:
2025-04-14T09:28:25.778000Z TestFramework (INFO): PRNG seed is: 2393500354933353065
2025-04-14T09:28:25.779000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20250414_092629/feature_bip68_sequence_271
2025-04-14T09:28:26.365000Z TestFramework (INFO): Running test disable flag
2025-04-14T09:28:26.430000Z TestFramework (INFO): Running test sequence-lock-confirmed-inputs
2025-04-14T09:29:22.254000Z TestFramework (INFO): Running test sequence-lock-unconfirmed-inputs
2025-04-14T09:29:23.869000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 182, in main
    self.run_test()
  File "/root/ci_scratch/build/test/functional/feature_bip68_sequence.py", line 73, in run_test
    self.test_sequence_lock_unconfirmed_inputs()
  File "/root/ci_scratch/build/test/functional/feature_bip68_sequence.py", line 336, in test_sequence_lock_unconfirmed_inputs
    assert_equal(None if i == 1 else 'inconclusive', self.nodes[0].submitblock(block.serialize().hex()))
  File "/root/ci_scratch/test/functional/test_framework/util.py", line 77, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(None == )
```

--- a/test/functional/test_framework/test_node.py
+++ b/test/functional/test_framework/test_node.py
@@ -919,6 +919,8 @@ class TestNodeCLI():
             # Ignore cli_stdout, raise with cli_stderr
             raise subprocess.CalledProcessError(returncode, p_args, output=cli_stderr)
         try:
+            if not cli_stdout.strip():
+                return None
             return json.loads(cli_stdout, parse_float=decimal.Decimal)
         except (json.JSONDecodeError, decimal.InvalidOperation):
             return cli_stdout.rstrip("\n")

